File: 프로그래머스/2/70129. 이진 변환 반복하기/이진 변환 반복하기.java
class Solution {
    public int[] solution(String s) {
        int[] answer = new int [2];
        
        while(s.length() > 1){
            int originlen = s.length();
            s = s.replaceAll("0","");
            
            answer[0]++;
            answer[1] += originlen - s.length();
            s = Integer.toBinaryString(s.length()); 
        }
        return answer;
    }
}

File: 프로그래머스/2/70129. 이진 변환 반복하기/이진 변환 반복하기.py
def solution(s):
    answer = []
    
    rm_zero = 0
    count = 0
    
    while s != '1':
        rm_zero += s.count('0')
        s = s.replace('0', '')
        count += 1
        s = bin(len(s))[2:]
        # print(s)
        
    return [count, rm_zero]